### ISSUE TITLE
Oultine seems broken in a C++ project with custom copilation

### ISSUE THREAD

**Issue by jclaveau** [eranif/codelite#699]
I just installed Codelite which seems to fit exactly my needs : PHP and C++ projects.
I have been able to configure my PHP projects exactly as expected and am really seduced by the result but concerning my C++ project the "Outline" panel stays empty (and quick outline doesn't work either).

So I downloaded Codelite from Github and recompile it without any better result.
Then I created an empty test project based on Qt and here the outline worked!

The concerned project is https://github.com/mixxxdj/mixxx/ which is a Qt4 based project compile with Scons. I configured it with the "customize" section of the project's settings like the following :
+ Build : scons -j 4
+ Clean : scons -c

The only kind of clue I found is this warning in the trace : 
Warning: Resource files must have same version number.

I dislike to file a bug for a program I don't know well, it's probably a misuse, but I think that it looks so unexplicit that, at least, I should warn the dev team.

Thanks in advance and thank you for your work anyway,
Jean Claveau

**Comment by eranif** [comment]
Thanks for the detailed description.

> The only kind of clue I found is this warning in the trace : 
> Warning: Resource files must have same version number.

This is just a warning that can be safely ignored

> The concerned project is https://github.com/mixxxdj/mixxx/ which is a Qt4 based project compile with Scons. 

The Outline mechanism, has nothing to do with the project type.
EDIT: 
I just cloned your repo, created an empty workspace, followed by a "Custom" project, imported all the files into CodeLite and it worked... 

( in codelite 7.1 you can simply drag the folder from the Explorer/Window Manager and drop it on the project )

Did you try to re-parse the workspace? `Workspace->Retag Workspace (full)`

![codelite-mixx](https://cloud.githubusercontent.com/assets/3677711/6652823/28afc9fa-ca85-11e4-9a1f-66b24efce092.png)

Eran

**Comment by eranif** [comment]
Also, which OS did you try to use?

**Comment by eranif** [comment]
Any update?
Eran

**Comment by jclaveau** [comment]
Sorry for the delay, heavy weekend ;)
I'll dig more after work. Thanks for the quick answers.

**Comment by jclaveau** [comment]
So,
To avoid any bad configuration from me I simply opened the same C++ file as you without creating a project before. The result is the same : No outline.

![codelite_outline_mixxx](https://cloud.githubusercontent.com/assets/1556489/6694107/1d541c46-ccd8-11e4-8342-8b209a33c186.png)

My os is Xubuntu 14.04.

Is there any way to get a more verbose log to send it to you?

**Comment by eranif** [comment]
Outline (as well as code completion and other C++ features) only works when a workspace exists
Try this (assuming you are using the latest git build):
* Close any opened workspace you have
* Open your favorite file manager and drag mixx folder and drop it on the "Workspace" tab

If done correctly, CodeLite will create a workspace with a default project and will import all source files. Once imported it will parse them and once this is done, you should have working environment to work with

Eran

**Comment by jclaveau** [comment]
It's alive!!! Thank you very much. Is there something I did wrong during the project creation?

Anyway, I just compiled the debug version to provide better reports in the future. Thank you very much for your extraquick answers and let's test CodeLite for C++ development!

**Comment by eranif** [comment]
I am not sure what went wrong in the process..
One point to remember though:
When creating a project using the "Drag and Drop" feature, CodeLite created a console project.
For your purposes, you need to change the project type:
`Project settings->Customize->Custom Build->Enable custom build`
and populate your `scons` commands there

**Comment by jclaveau** [comment]
It's already done, thanks :)
Another good thing, as it's a qt4 project, is to add /usr/include/qt4/ for autocomplete on my OS.

I'll dig deeper and try to report as much thing I can.

See you soon I guess

**Comment by eranif** [comment]
Actually, if you are using Qt4, then you might want to add the following paths:
(assuming you are using statements like this `#include <QString>`)
```
/usr/include/qt4/QtCore
/usr/include/qt4/QtXml
/usr/include/qt4/QtSql
```
I am not sure if these paths exist on your system, but you get the idea 
You should add these paths under `settings->code completion->ctags->search paths`
